erpnext/accounts/doctype/payment_request/payment_request.js

```diff
@@ -49,3 +49,20 @@ frappe.ui.form.on("Payment Request", "refresh", function(frm) {
 	}
 });
 
+frappe.ui.form.on("Payment Request", "is_a_subscription", function(frm) {
+	frm.toggle_reqd("payment_gateway_account", frm.doc.is_a_subscription);
+	frm.toggle_reqd("subscription_plans", frm.doc.is_a_subscription);
+
+	if (frm.doc.is_a_subscription) {
+		frappe.call({
+			method: "get_subscription_details",
+			doc: frm.doc,
```

Comment:
sending frm.doc might be expensive, can we add get_subscription_details as non-class method?

Reply:
Hi @saurabh6790,

Thanks a lot for your review.

If we send the reference doctype and reference name as parameters in a non-class method, we can get the plans back and we insert them back using javascript ? Would that be less expensive ?
I'll propose a new solution.

Reply:
Yeah! Just send the required parameters to the method. Sending the whole `frm` in a request is expensive.

Reply:
Thanks for your answer!
I have pushed a new proposal, please let me know if that's what you had in mind please.